### PR TITLE
Adjust couple of tests for aarch64

### PR DESCRIPTION
**Pull request by vdimir**
Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Ref https://github.com/ClickHouse/ClickHouse/issues/33716

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/queries/0_stateless/01492_format_readable_quantity.sql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-WITH round(exp(number), 6) AS x, toUInt64(x) AS y, toInt32(x) AS z
+WITH round(exp(number), 6) AS x, toUInt64(x) AS y, toInt32(min2(x, 2147483647)) AS z
```
Comment (vdimir) [inline tests/queries/0_stateless/01492_format_readable_quantity.sql:1]: Different overflow behavior

File: tests/queries/0_stateless/02207_s3_content_type.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 . "$CURDIR"/../shell_config.sh
 
 $CLICKHOUSE_CLIENT --query "
-INSERT INTO TABLE FUNCTION s3('http://localhost:11111/test/content-type.csv.gz', 'test', 'testtest', 'CSV', 'number UInt64') SELECT number FROM numbers(1000000) SETTINGS s3_max_single_part_upload_size = 10000, s3_truncate_on_insert = 1;
```
Comment (vdimir) [inline tests/queries/0_stateless/02207_s3_content_type.sh:10]: gzip works a bit differently (didn't investigate much, but both files successfully decompressed with `gunzip`), size of plain file is more predictable

File: tests/queries/0_stateless/01854_HTTP_dict_decompression.python [hunk 3]
```diff
@@ -158,7 +158,7 @@ def test_select(dict_name="", schema="word String, counter UInt32", requests=[],
 
         COMPRESS_METHOD = requests[i]
         print(i, COMPRESS_METHOD, ADDING_ENDING, SEND_ENCODING)
-        check_answers("select * from {}".format(dict_name), answers[i])
+        check_answers("SELECT * FROM {} ORDER BY word".format(dict_name), answers[i])
```
Comment (vdimir) [inline tests/queries/0_stateless/01854_HTTP_dict_decompression.python:161]: Different order of rows

File: tests/queries/0_stateless/01103_check_cpu_instructions_at_startup.sh [hunk 4]
```diff
@@ -10,6 +10,8 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 # Don't even try to do that. This test should be disabled for sanitizer builds.
 ${CLICKHOUSE_LOCAL} --query "SELECT max(value LIKE '%sanitize%') FROM system.build_options" | grep -q '1' && echo '@@SKIP@@: Sanitizer build' && exit
 
+${CLICKHOUSE_LOCAL} --query "SELECT value FROM system.build_options WHERE name = 'SYSTEM_PROCESSOR';" | grep -q 'aarch64' && echo '@@SKIP@@: aarch64 build' && exit
```
Comment (vdimir) [inline tests/queries/0_stateless/01103_check_cpu_instructions_at_startup.sh:13]: Skip this test on arm

File: tests/queries/0_stateless/01016_simhash_minhash.sql [hunk 5]
```diff
@@ -75,38 +75,38 @@ SELECT 'uniqExact', uniqExact(s) FROM defaults;
 
 
 SELECT 'ngramSimHash';
-SELECT arrayStringConcat(groupArray(s), '\n:::::::\n'), count(), ngramSimHash(s) as h FROM defaults GROUP BY h;
+SELECT arrayStringConcat(groupArray(s), '\n:::::::\n'), count(), ngramSimHash(s) as h FROM defaults GROUP BY h ORDER BY h;
```
Comment (vdimir) [inline tests/queries/0_stateless/01016_simhash_minhash.sql:78]: Different order of rows

**Review by vdimir — COMMENTED** [review]
Added small comments with explanations about changes

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**